play.pokemonshowdown.com/src/client-main.ts

```diff
@@ -373,6 +387,69 @@ class PSTeams extends PSStreamModel<'team' | 'format'> {
 			key: '',
 		};
 	}
+	loadRemoteTeams() {
+		PSLoginServer.query('getteams', {}).then(data => {
+			if (!data) return;
+			if (data.actionerror) {
+				return PS.alert('Error loading uploaded teams: ' + data.actionerror);
+			}
+			for (const team of data.teams as ((Team & { team: string })[])) {
+				let matched = false;
+				for (const curTeam of this.list) {
+					let match = this.compareTeams(team, curTeam);
+					if (match === true) {
+						// prioritize locally saved teams over remote
+						// as so to not overwrite changes
+						matched = true;
+						break;
+					}
+					if (match === 'rename') {
+						delete curTeam.teamid;
+						if (!team.name.endsWith(' (server version)')) {
+							team.name += ' (server version)';
+						}
+					}
+				}
+				team.loaded = false;
+				if (!matched) {
+					// hack so that it shows up in the format selector list
+					team.folder = '';
+					// team comes down from loginserver as comma-separated list of mons
+					// to save bandwidth
+					let mons = team.team.split(',').map((m: string) => ({ species: m, moves: [] }));
+					team.packedTeam = PSTeambuilder.packTeam(mons);
+					team.iconCache = null;
+					this.push(team);
+				}
+			}
+		});
+	}
+	compareTeams(serverTeam: Team & { team: string }, localTeam: Team) {
+		// if titles match exactly and mons are the same, assume they're the same team
+		// if they don't match, it might be edited, but we'll go ahead and add it to the user's
+		// teambuilder since they may want that old version around. just go ahead and edit the name
+		const mons = serverTeam.team.split(',');
+		let matches = 0;
+		const otherMons = PSTeambuilder.unpackTeam(localTeam.packedTeam);
+		for (const mon of mons) {
+			for (const otherMon of otherMons) {
+				if (toID(otherMon.species) === toID(mon)) {
+					matches++;
+					break;
+				}
+			}
+		}
+		let sanitize = (name: string) => (name || "").replace(/\s+\(server version\)/g, '').trim();
+		const nameMatches = sanitize(serverTeam.name) === sanitize(localTeam.name);
+		if (!(nameMatches && serverTeam.format === localTeam.format)) {
+			return false;
+		}
+		// if it's been edited since, invalidate the team id on this one (count it as new)
+		// and load from server
+		if (mons.length !== otherMons.length || matches !== otherMons.length) return 'rename';
```

Comment:
```suggestion
		let sanitize = (name: string) => (name || "").replace(/\s+\(server version\)/g, '').trim();
		const nameMatches = sanitize(serverTeam.name) === sanitize(localTeam.name);
		if (!(nameMatches && serverTeam.format === localTeam.format)) {
			return false;
		}
		// if it's been edited since, invalidate the team id on this one (count it as new)
		// and load from server
		const mons = serverTeam.team.split(',').map(toID).sort().join(',');
		const otherMons = PSTeambuilder.unpackTeam(localTeam.packedTeam).map(p => toID(p.species)).sort().join(',');
		if (mons !== otherMons) return 'rename';
```